substrate/frame/nomination-pools/src/migration.rs

```diff
@@ -725,3 +725,126 @@ pub mod v5 {
 		}
 	}
 }
+
+/// This migration accumulates and initializes the [`TotalValueLocked`] for all pools.
+pub mod v6 {
+	use super::*;
+
+	/// [`VersionUncheckedMigrateV5ToV6`] wrapped in a
+	/// [`frame_support::migrations::VersionedMigration`], ensuring the migration is only
+	/// performed when on-chain version is 5.
+	#[cfg(feature = "experimental")]
+	pub type VersionCheckedMigrateV5ToV6<T> = frame_support::migrations::VersionedMigration<
+		5,
+		6,
+		VersionUncheckedMigrateV5ToV6<T>,
+		crate::pallet::Pallet<T>,
+		<T as frame_system::Config>::DbWeight,
+	>;
+
+	pub struct VersionUncheckedMigrateV5ToV6<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> VersionUncheckedMigrateV5ToV6<T> {
+		fn calculate_tvl_by_total_stake() -> BalanceOf<T> {
+			BondedPools::<T>::iter()
+				.map(|(id, inner)| {
+					T::Staking::total_stake(
+						&BondedPool { id, inner: inner.clone() }.bonded_account(),
+					)
+					.unwrap_or_default()
+				})
+				.reduce(|acc, total_balance| acc + total_balance)
+				.unwrap_or_default()
+		}
+	}
+
+	impl<T: Config> OnRuntimeUpgrade for VersionUncheckedMigrateV5ToV6<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let migrated = BondedPools::<T>::count();
+			// The TVL should be the sum of all the funds that are actively staked and in the
+			// unbonding process of the account of each pool.
+			let tvl: BalanceOf<T> = Self::calculate_tvl_by_total_stake();
+
+			TotalValueLocked::<T>::set(tvl);
+
+			log!(info, "Upgraded {} pools with a TVL of {:?}", migrated, tvl);
+
+			// reads: migrated * (BondedPools +  Staking::total_stake) + count + onchain
+			// version
+			//
+			// writes: current version + TVL
+			T::DbWeight::get().reads_writes(migrated.saturating_mul(2).saturating_add(2).into(), 2)
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
+			ensure!(
```

Comment:
interesting. Harmless to have these, but what caused you to think you might mess with the decoding of these? the migration is only writing to `TotalValueLocked` after all, the rest is just reads.

Reply:
Yeah I think the below checks can be removed. The last migration was I think migrating one of the storage items and had these checks just to ensure it does not mess up anything. This migration does not touch any existing storages.

Reply:
Optionally these checks could be added to `try_state` if its not already to ensure all values are decodedable.